src/compiler/parser.ts

```diff
@@ -3225,6 +3228,16 @@ namespace ts {
             }
         }
 
+        function parseAssertsTypePredicate(): TypeNode {
+            const node = <TypePredicateNode>createNode(SyntaxKind.TypePredicate);
+            node.assertsModifier = parseExpectedToken(SyntaxKind.AssertsKeyword);
+            node.parameterName = parseIdentifier();
+            if (parseOptional(SyntaxKind.IsKeyword)) {
+                node.type = parseType();

```

Comment:
This makes this type of object polymorphic. Could you instead always assign the property and use `undefined` if there is no type?

Reply:
Yup